src/transformers/generation/tf_utils.py

```diff
@@ -2009,16 +2014,22 @@ def beam_search_cond_fn(
             not_max_length_yet = cur_len < max_length
 
             # 2. can the new beams still improve?
-            best_running_score = running_scores[:, :1] / (max_length**length_penalty)
+            # early_stopping == False -> apply heuristic = always get the best score from `cur_len`
```

Comment:
can't comment at the exact line, but the comment above `def beam_search_cond_fn` contains `define stop-condition`. I am always confused by this comment, as well as the code logic here between PT/TF.

- The comment should be `continuation condition` instead of `stop condition`
- Not sure why we choose to implement `continuation` condition in TF but `stop` condition in PT. It would be much easier if both implement the same thing - at least much easier to debug if some inconsistency occur between PT/TF.
  - I am not asking for a change (at least not in this PR), just a general comment

Reply:
I'd say it's mostly semantics, `continue` = `not stop` :p Stopping condition is a common term used in iterative methods (like generate), hence the name.

On the TF/FLAX side, the loop function actually requires a "continuation condition" (as any while function in programming)